### ISSUE TITLE
[mpi]Inconsistencies for distributed wells.

### ISSUE THREAD

**Issue by blattms** [OPM/opm-simulators#3931]
We are facing deadlocks with distributed wells when calling `BlackoilModel::prepareTimeStep()`. Worked with last year's release. Still need to dig in to find when and how this actually set or computed (any hint appreciated). It seems like this was implemented assuming that all preforations are on the same process. At least if we artificially force wells to be distributed on Norne this is not the case for C-4H

**Comment by blattms** [comment]
Some how it seems that events at [BlackoilWellModel_impl.hpp#L1492](/OPM/opm-simulators/blob/57d8f1aa244cf08526f186db6df54260d830eeae/opm/simulators/wells/BlackoilWellModel_impl.hpp#L1492) are inconsistent (for whatever reason). On one process:
```
(gdb) p events
$3 = (Opm::Events &) @0x5555693ad5f8: {m_events = 93825124556720}
```
on the other:
```
 p events
$3 = (Opm::Events &) @0x5555675c6330: {m_events = 257}
```

**Comment by blattms** [comment]
resolved via #3983
